libfwupdplugin/fu-udev-device.c

```diff
@@ -2450,44 +2450,40 @@ FuDevice *
 fu_udev_device_find_usb_device(FuUdevDevice *self, GError **error)
 {
 #if defined(HAVE_GUDEV) && defined(HAVE_LIBUSB)
-	FuUdevDevicePrivate *priv = GET_PRIVATE(self);
-	guint8 bus = 0;
-	guint8 address = 0;
+	guint64 bus = 0;
+	guint64 address = 0;
 	gint rc;
 	libusb_device **dev_list = NULL;
-	g_autoptr(GUdevDevice) udev_device = NULL;
+	g_autofree gchar *attr_bus = NULL;
+	g_autofree gchar *attr_dev = NULL;
+	g_autoptr(FuUdevDevice) udev_device = NULL;
 	g_autoptr(FuUsbDevice) usb_device = NULL;
 	g_autoptr(libusb_context) ctx = NULL;
 
 	g_return_val_if_fail(FU_IS_UDEV_DEVICE(self), NULL);
 	g_return_val_if_fail(error == NULL || *error == NULL, NULL);
 
-	/* sanity check */
-	if (priv->udev_device == NULL) {
-		g_set_error_literal(error, FWUPD_ERROR, FWUPD_ERROR_NOT_FOUND, "not initialized");
-		return NULL;
-	}
-
 	/* look at the current device and all the parent devices until we can find the USB data */
-	udev_device = g_object_ref(priv->udev_device);
-	while (udev_device != NULL) {
-		g_autoptr(GUdevDevice) udev_device_parent = NULL;
-		bus = g_udev_device_get_sysfs_attr_as_int(udev_device, "busnum");
-		address = g_udev_device_get_sysfs_attr_as_int(udev_device, "devnum");
-		if (bus != 0 || address != 0)
-			break;
-		udev_device_parent = g_udev_device_get_parent(udev_device);
-		g_set_object(&udev_device, udev_device_parent);
-	}
+	udev_device = fu_udev_device_get_parent_with_subsystem(self, "usb", "usb_device", NULL);
```

Comment:
aren't you missing the `error` argument here?

Reply:
Am indeed, good catch! Fixed.